integration_tests/src/main/python/cache_test.py

```diff
@@ -234,3 +234,41 @@ def write_read_parquet_cached(spark):
           'spark.sql.parquet.outputTimestampType': ts_write}
 
     assert_gpu_and_cpu_are_equal_collect(read_parquet_cached(data_path_gpu), conf)
+
+@pytest.mark.parametrize('enableVectorized', ['false', 'true'], ids=idfn)
+@pytest.mark.parametrize('func', [with_cpu_session, with_gpu_session])
+@allow_non_gpu("ProjectExec", "Alias", "Literal", "DateAddInterval", "MakeInterval", "Cast",
+               "ExtractIntervalYears", "Year", "Month", "Second", "ExtractIntervalMonths",
+               "ExtractIntervalSeconds", "SecondWithFraction", "ColumnarToRowExec")
+@pytest.mark.parametrize('selectExpr', [("NULL as d", "d"),
+                                        # We need to get individual elements of interval because pyspark doesn't know how # to parse CalendarIntervalType
+                                        ("make_interval(0,m,0,d,0,0,s) as d", ["cast(extract(years from d) as long)", "extract(months from d)", "extract(seconds from d)"])])
```

Comment:
Then perhaps this should be a scala test instead of a python test.

Reply:
I didn't think that was necessary because I was checking the elements of the interval which depend on what was originally set? I also am checking the cached and non-cached versions. Please let me know if you still think we want this to be a scala test. I will be happy to make the change, it should be easy

Reply:
I just thought it would be simpler to fully test the code if you could get proper support for the type. If you want to leave it in that is okay. My main issue is that we are testing only a single value, where all of the values are 0.  It would be good to have a way to test negative values, null calendar intervals, etc. to be sure that we have hit the various corner cases.